### PR TITLE
magickload: add support for cursor images

### PR DESCRIPTION
**Pull request by marcosc90**
This PR adds support for cursor images which are a type of ICO files, see https://docs.fileformat.com/image/ico/#header

## Header
Offset | Size (in bytes) | Purpose
--- | --- | ---
0 | 2 | Reserved. Must always be 0.
2 | 2 | Specifies image type: 1 for icon (.ICO) image, 2 for cursor (.CUR) image. Other values are invalid.
4 | 2 | Specifies number of images in the file.

---

```
vips copy sample.cur out.png
```

No longer throws:

```
magickload: Magick: improper image header `sample.cur' @ error/tga.c/ReadTGAImage/253 (null)
```

---
Sample: http://www.rw-designer.com/cursor-detail/193983
License: Public Domain [license](http://www.rw-designer.com/licenses).

### CODE REVIEW

File: libvips/foreign/magick.c [hunk 1]
```diff
@@ -83,8 +83,9 @@ magick_sniff(const unsigned char *bytes, size_t length)
 	if (length >= 4 &&
 		bytes[0] == 0 &&
 		bytes[1] == 0 &&
-		bytes[2] == 1 &&
-		bytes[3] == 0)
+		(bytes[2] == 1 || bytes[2] == 2) &&
+		bytes[3] == 0 &&
+		!(bytes[4] == 0 && bytes[5] == 0))
```
Comment (kleisauke) [inline libvips/foreign/magick.c:88]: nit: I would write this as:
```suggestion
		(bytes[4] != 0 || bytes[5] != 0)
```

**Review by kleisauke — COMMENTED** [review]
LGTM % comments

File: libvips/foreign/magick.c [hunk 2]
```diff
@@ -80,11 +80,12 @@
...
 {
-	if (length >= 4 &&
+	if (length >= 6 &&
 		bytes[0] == 0 &&
 		bytes[1] == 0 &&
-		bytes[2] == 1 &&
-		bytes[3] == 0)
+		(bytes[2] == 1 || bytes[2] == 2) &&
+		bytes[3] == 0 &&
+		(bytes[4] != 0 || bytes[5] != 0)
```
Comment (kleisauke) [inline libvips/foreign/magick.c:88]: Sorry, I accidentally removed a `)` in my previous code suggestion.
```suggestion
		(bytes[4] != 0 || bytes[5] != 0))
```

Comment (marcosc90) [reply libvips/foreign/magick.c:88]: Yeah no worries was about to push the fix

**Review by kleisauke — APPROVED** [review]

**Pull request merged**